runtime/drivers/clickhouse/olap.go

```diff
@@ -503,7 +505,55 @@ func (c *connection) renameTable(ctx context.Context, oldName, newName, onCluste
 }
 
 func (c *connection) MayBeScaledToZero(ctx context.Context) bool {
-	return c.config.CanScaleToZero
+	if c.config.APIKeyID == "" {
+		// no api key provided resort to the config set
+		return c.config.CanScaleToZero
+	}
+
+	c.statusCheckMutex.Lock()
+	defer c.statusCheckMutex.Unlock()
+	// check if stauts is cached
+	if !c.statusCheckedAt.IsZero() && time.Since(c.statusCheckedAt) <= time.Minute*10 {
+		return c.scaledToZero
+	}
```

Comment:
I think caching for 10 mins is too long, makes it too likely to lead to a feedback loop that keeps it alive